### PR TITLE
decode_b: Make b_dim safe

### PR DESCRIPTION
**Pull request by randomPoison**

### CODE REVIEW

File: src/decode.rs [hunk 1]
```diff
@@ -7956,9 +7950,7 @@ unsafe fn decode_b(
             1 => {
                 (*(&mut *(t.l.tx_intra).as_mut_ptr().offset(by4 as isize)
                     as *mut int8_t as *mut alias8))
-                    .u8_0 = (0x1 as libc::c_int
-                    * *b_dim.offset((2 as libc::c_int + 1 as libc::c_int) as isize)
-                        as libc::c_int) as uint8_t;
+                    .u8_0 = (0x1 * b_dim[2 + 1] as libc::c_int) as uint8_t;
```
Comment (kkysen) [inline src/decode.rs:7953]: What's the point of `0x1 * ` here?  I think the casts are redundant, going from `u8` to `c_int` to `u8` again.

```suggestion
                    .u8_0 = b_dim[2 + 1];
```

Comment (kkysen) [reply src/decode.rs:7953]: Oh, I guess the other ones below are multiplied by different things, so this is to keep parity.  We can leave this as is then, and clean it up more later.

File: src/decode.rs [hunk 2]
```diff
@@ -12934,7 +12864,7 @@ unsafe fn decode_b(
                         t,
                         lowest_px,
                         0 as libc::c_int,
-                        b_dim,
+                        b_dim.as_ptr(),
```
Comment (kkysen) [inline src/decode.rs:12867]: Didn't I change `obmc_lowest_px` to take `b_dim` as `&[u8; 4]` and not a pointer already?  Or is this PR just branched off before I made that change?

**Review by kkysen — APPROVED** [review]

File: src/decode.rs (hunk 2 above)
Comment (randomPoison) [reply src/decode.rs]: Yup, that's fixed up after a rebase.

**Pull request merged**